engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery.html

```diff
@@ -1,86 +1,135 @@
+      <label class="col-sm-2 control-label" for="urlToDiscover">URL to Discover</label>
+      <div class="col-sm-6">
+        <input id="urlToDiscover"
+               type="text"
+               class="form-control"
+               ng-pattern="/^(file:\/\/|http:\/\/|https:\/\/).+/"
+               ng-model="discovery.url"
+               ng-disabled="discovery.working"
+               name="discoveryUrl"
+               required/>
+      </div>
+    </div>
 
-      <span class="input-group-btn input-group-lg">
-        <button ng-show="!discovery.pending && permitted('edit_products')"
-               ng-disabled="discoveryForm.discoveryUrl.$invalid"
-               translate
-               type="submit"
-               class="btn btn-primary btn-lg">
-          Discover
-        </button>
+    <div class="form-group" ng-show="discovery.contentType === 'docker'">
+      <label class="col-sm-2 control-label" for="registryToDiscover">Registry to Discover</label>
+      <div class="col-sm-6">
+        <input id="registryToDiscover"
+               type="text"
+               class="form-control"
+               ng-model="discovery.url"
+               ng-disabled="discovery.working"
+               placeholder="Example: registry.access.redhat.com or https://registry.access.redhat.com"
+               name="discoveryUrl"
+               required/>
+      </div>
+    </div>
 
-        <button type="button"
-                ng-disabled="discovery.working"
-                ng-show="discovery.pending && permitted('edit_products')"
-                ng-click="cancelDiscovery()"
-                class="btn btn-primary btn-lg">
-          <i class="fa fa-spinner fa-spin" ng-show="discovery.pending"></i>
-          <span translate>Cancel</span>
-        </button>
-      </span>
+    <div class="form-group" ng-show="discovery.contentType === 'docker'">
+      <label class="col-sm-2 control-label" for="registryUsername">Registry Username</label>
+      <div class="col-sm-6">
+        <input id="upstreamUsername"
+               type="text"
+               class="form-control"
+               ng-model="discovery.upstreamUsername"
+               ng-disabled="discovery.working"
+               name="upstreamUsername"/>
+      </div>
+    </div>
 
+    <div class="form-group" ng-show="discovery.contentType === 'docker'">
+      <label class="col-sm-2 control-label" for="registryPassword">Registry Password</label>
+      <div class="col-sm-6">
+        <input id="upstreamPassword"
+               type="password"
+               class="form-control"
+               ng-model="discovery.upstreamPassword"
+               ng-disabled="discovery.working"
+               name="upstreamPassword"/>
+      </div>
     </div>
-  </form>
-</div>
 
-<section>
-  <div class="row">
-    <h3 class="col-sm-12" translate>Results</h3>
+    <div class="form-group">
+      <label class="col-sm-2 control-label"></label>
```

Comment:
This was to align the Discover button, should it not be shoved to the right?
![repository discovery](https://cloud.githubusercontent.com/assets/338096/23520829/81bec7d4-ff4a-11e6-902b-ca47b80b1aac.png)